pkg/headerutil/headerutil.go

```diff
@@ -0,0 +1,46 @@
+package headerutil
+
+import (
+	"strconv"
+	"strings"
+)
+
+type Range struct {
+	// If provided, specifies field to sort by.
+	Sort *string
+
+	// If provided, limits the results to the provided value.
+	Max *int
+
+	// The order the results are returned in.
+	Order *string
+}
+
+func ParseRange(header string) (*Range, error) {
```

Comment:
Could we add some unit tests for this. Probably just some table driven tests with a bunch of various inputs.